airflow/api_fastapi/common/db/dags.py

```diff
@@ -17,31 +17,40 @@
 
 from __future__ import annotations
 
+from typing import TYPE_CHECKING
+
 from sqlalchemy import func, select
 
+if TYPE_CHECKING:
+    from sqlalchemy.sql import Select
+
 from airflow.models.dag import DagModel
 from airflow.models.dagrun import DagRun
 
-latest_dag_run_per_dag_id_cte = (
-    select(DagRun.dag_id, func.max(DagRun.start_date).label("start_date"))
-    .where()
-    .group_by(DagRun.dag_id)
-    .cte()
-)
 
-
-dags_select_with_latest_dag_run = (
-    select(DagModel)
-    .join(
-        latest_dag_run_per_dag_id_cte,
-        DagModel.dag_id == latest_dag_run_per_dag_id_cte.c.dag_id,
-        isouter=True,
+def generate_dag_select_query(
+    stmt: Select = select(DagRun).where().cte(), use_outer_join: bool = True
+) -> Select:
```

Comment:
```suggestion
def generate_dag_select_query(
    stmt: Select | None = None, use_outer_join: bool = True
) -> Select:
    if not stmt:
        stmt = select(DagRun).where().cte()
```
I think this should be better or removing the stmt arg. WDYT?

Reply:
+1 for that.

Reply:
Modified the definition and changed the name.